src/experimental/tools/download2go/apis/dash/dashConnectivity.ts

```diff
@@ -0,0 +1,188 @@
+ * limitations under the License.
+ */
+
+import { Observable, of, throwError } from "rxjs";
+import { mergeMap } from "rxjs/operators";
+
+import { ISegment } from "../../../../../manifest";
+import { ISidxSegment } from "../../../../../parsers/containers/isobmff";
+import parseManifest from "../../../../../parsers/manifest/dash/index";
+import { IParserResponse } from "../../../../../parsers/manifest/dash/parse_mpd";
+import { IParsedManifest } from "../../../../../parsers/manifest/types";
+import xhrRequest from "../../../../../utils/request";
+import { makeHTTPRequest, SegmentConstuctionError } from "../../utils";
+import { ITypedArray } from "../drm/keySystems";
+import { IUtils } from "./../../types";
+import {
+  IRepresentation,
+  ISegmentBuilder,
+  ISegmentBuilt,
+  ISegmentsBuiltType,
+} from "./types";
+
+/**
+ * Get a manifest from a server and parse it.
+ *
+ * @param url - The url manifest mpd file
+ * @returns The manifest parsed IParserResponse<IParsedManifest>
+ *
+ */
+export const getOnlineMPDParsed = async (
+  url: string
+): Promise<IParserResponse<IParsedManifest>> => {
+  const data = await makeHTTPRequest<string>(url, {
+    method: "GET",
+    responseType: "text",
+  });
+  return parseManifest(new DOMParser().parseFromString(data, "text/xml"), {
+    loadExternalClock: false,
+    url,
+  });
+};
+
+/**
+ * Build the init segments to be able to retieve SIDX (index/init)
+ *
+ * @param IRepresentation - The segmentBase and initialization segment
+ * @returns The initSegment and indexSegment
+ *
+ */
+export const buildInitIndexSegment = async ({
+  segmentBase,
+  initialization,
+}: IRepresentation): Promise<{
+initSegment: ITypedArray | ArrayBuffer;
+indexSegment: ITypedArray | ArrayBuffer;
+}> => {
+  try {
+    if (!initialization.mediaURL) {
+      throw new Error("MediaURL from the initialization segment is broken");
+    }
+    const [initSegment, indexSegment] = await Promise.all([
+      makeHTTPRequest<ITypedArray | ArrayBuffer>(initialization.mediaURL, {
+        headers: { Range: `bytes=${initialization.range.join("-")}` },
+        method: "GET",
+        responseType: "arraybuffer",
+      }),
+      makeHTTPRequest<ITypedArray | ArrayBuffer>(initialization.mediaURL, {
+        headers: { Range: `bytes=${segmentBase.indexRange.join("-")}` },
+        method: "GET",
+        responseType: "arraybuffer",
+      }),
+    ]);
+    return { initSegment, indexSegment };
+  } catch (e) {
+    throw new SegmentConstuctionError(e.message);
+  }
+};
+
+/**
+ * Get a segment from a server.
+ *
+ * @param ISidxSegment - SIDX or ISegment Segment information
+ * @param url - base url of the segment
+ * @returns The data,duration,time,timescale of the current segment retrieved
+ *
+ */
+export const getSegmentBuffer = ({
+  segment,
+  url,
+  type,
+}: {
+segment: ISidxSegment | ISegment;
+url: string | null;
+type: "TemplateRepresentationIndex" | "BaseRepresentationIndex";
```

Comment:
You should not need to have that information:
  - either you have an `indexRange` in the init segment in which case you have to request the sidx/webm cues
  - either you do not

And:
  - either each segment has a `range` in which case you have to perform byte range requests
  - either they do not, in which case you load the whole URL

Reply:
Ok!

I will get rid of the type and then use `indexRange`/`range` parameter.